### ISSUE TITLE
ncplane_set_bg_palindex seems to be setting style bits

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#2049]
I'm working on converting `ncneofetch` to CLI mode from direct mode (see #2030), and running into a problem with `ncplane_set_*_palindex()`. The background version seems to be setting style bits. The foreground version seems to set everything to black. Odd!

**Comment by dankamongmen** [comment]
```c
int ncplane_set_fg_palindex(ncplane* n, int idx){                                                                         
  if(idx < 0 || idx >= NCPALETTESIZE){                                                                                    
    return -1;                                                                                                            
  }                                                                                                                       
  n->channels |= NC_FGDEFAULT_MASK;                                                                                       
  n->channels |= NC_FG_PALETTE;                                                                                           
  ncchannels_set_fg_alpha(&n->channels, NCALPHA_OPAQUE);                                                                  
  n->stylemask &= 0xffff00ff;                                                                                             
  n->stylemask |= (idx << 8u);                                                                                            
  return 0;                                                                                                               
}                                                                                                                         
                                                                                                                          
int ncplane_set_bg_palindex(ncplane* n, int idx){                                                                         
  if(idx < 0 || idx >= NCPALETTESIZE){                                                                                    
    return -1;                                                                                                            
  }                                                                                                                       
  n->channels |= NC_BGDEFAULT_MASK;                                                                                       
  n->channels |= NC_BG_PALETTE;                                                                                           
  ncchannels_set_bg_alpha(&n->channels, NCALPHA_OPAQUE);                                                                  
  n->stylemask &= 0xffffff00;                                                                                             
  n->stylemask |= idx;                                                                                                    
  return 0;                                                                                                               
}                 
```

why are these screwing with `stylemask` at all?!

**Comment by dankamongmen** [comment]
i replaced these internals with calls to `ncchannels_set_*_palindex()`, and everything now works as expected. i guess these have just been broken since their inception, eh? sheesh.

**Comment by dankamongmen** [comment]
fixed for 2.3.14.
